src/main/java/service/PatientImpl.java

```diff
@@ -23,19 +31,37 @@ public static synchronized PatientImpl getInstance() {
 
   @Override
   public synchronized boolean bookAppointment(
-      PatientId patientId, AppointmentId appointmentId, AppointmentType type)
-      throws RemoteException {
-    return false;
+          PatientId patientId, AppointmentId appointmentId, AppointmentType type)
+          throws RemoteException {
+
+    Optional<Appointment> appointment = database.findByTypeAndId(type, appointmentId);
+    ArrayList<PatientId> patientIds = appointment.get().addPatient(patientId);
+    int capacity = appointment.get().getRemainingCapacity();
+    database.update(new Appointment(appointmentId, capacity, patientIds), appointmentId);
+
+    logger.info(
+            "Book appointment success: %s, %s"
+                    .formatted(type.toString(), appointmentId.getId()));
+    return true;
   }
 
   @Override
   public List<Appointment> getAppointmentSchedule(PatientId patientId) throws RemoteException {
-    return null;
+    return database.findAllByPatientId(patientId);
   }
 
   @Override
   public synchronized boolean cancelAppointment(PatientId patientId, AppointmentId appointmentId)
-      throws RemoteException {
-    return false;
+          throws RemoteException {
+    List<Appointment> appointmentList = database.findAllByPatientId(patientId);
```

Comment:
I think we need to add an AppointmentType parameter. An appointmentId itself can not uniquely identify an appointment. 
Then, consider doing in this way:
1. find appointment by type and id
2. remove the patient id from the patientIds in the appointment

Reply:
OK. 
Then you can iterate over the types and try to look for the id in each type. If you find the id in a type, then you can get the appointment by the id and make the change.
So no need to get all appointments with a patient

Reply:
professional